### ISSUE TITLE
Allow usage of multiple screens

### ISSUE THREAD

**Issue by GoogleCodeExporter** [brockgr/csshx#18]
```
As nice as the master input window is, it takes up a lot of space. And when 
you're working on a 12 
box cluster, the allowed terminal space is uncomfortably small.

It would be nice to see support for allowing the grid to span 2, or even 3 
screens. I think this should 
NOT be a default, but if csshX is activated via something like --screen 0,1 the 
terminals should be 
allowed to grid using both screens (or any/all configured screens.

I do NOT feel as though terminals should be split by the part between monitors. 
Meaning don't split 
a single terminal in half and have each part reside on each screen.
```

Original issue reported on code.google.com by `vxjaso...@gmail.com` on 17 Jul 2009 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
I think this is a good idea. I will look into it when I get hold of a second 
monitor.
```

Original comment by `gavin.brock` on 29 Jul 2009 at 3:58
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
Original comment by `gavin.brock` on 29 Jul 2009 at 3:58
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
I actually use csshX with 25+ nodes clusters and a 13,3" screen so I use a 
second screen (20") but when I 
retile windows they all go back to the small screen... and I'm not sure of 
setting up my second monitor as the 
main screen.

I subscribe to this request and please count on me for testing/debugging new 
releases including this feature 
if you need.

Regards
Juan
```

Original comment by `mor...@gmail.com` on 12 Sep 2009 at 10:36

**Comment by GoogleCodeExporter** [comment]
```
I was having this same issue and this was how I dealt with it.

1. open csshx connection
2. ctrl-a
3. press b
4. drag the blue overlay to span all 2/3/4/etc. monitors
5. press enter in the blue window

That re-tiles all of my slave windows across multiple monitors and works like a 
champ.
```

Original comment by `chadilla...@gmail.com` on 2 Oct 2009 at 1:19

**Comment by GoogleCodeExporter** [comment]
```
Nice workaround. It'd be great to start up already configured, still.
```

Original comment by `vxjaso...@gmail.com` on 2 Oct 2009 at 5:40

**Comment by GoogleCodeExporter** [comment]
```
agreed, but it works great for now ;)
```

Original comment by `chadilla...@gmail.com` on 2 Oct 2009 at 5:27

**Comment by GoogleCodeExporter** [comment]
```
Was just messing with my config a little bit, found a bit of a bug using the 
method
above.

If you're primary desktop resides to the right side of your secondary monitor it
seems to cause the tiling to much up

1. primary (has Apple menu bar)
2. secondary

[1][2] = works fine
[2][1] = tiles windows on 1 but stacks them because the sizing accounts for 
extra
desktop but can't access it when tiling causing them to overlap.

The fix is setting your left most monitor as your "primary" as it seems to work 
fine
tiling to the right.
```

Original comment by `chadilla...@gmail.com` on 2 Oct 2009 at 5:44

**Comment by GoogleCodeExporter** [comment]
```
This issue has sat here for a little too long - I guess I'm not sure what is 
really needed.

Comment 4 is, I think, the recommended solution. If you drag the blue 'bounds' 
window to the desired size and position, you can 
press "p" to display the bounds that you are using (should be a string like 
{x,y,w,h}).

If you then paste that into the following line to your .csshrc, all future 
sessions should open with those bounds by default.

     screen_bounds = {x,y,w,h}

If you feel some other feature would help, please reopen this issue.

Thanks!! :-)

```

Original comment by `gavin.brock` on 23 Apr 2010 at 3:59
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
Ok - I had a rethink about this, and added basic support for multiple screens. 
You can now enter, instead of a single screen id, a screen range to use:

    csshX --screen 1-2 host1 host2 ...

The script will then try to guess a good rectangle that fits between those two 
screens.

This should work across more than two screens as well (you specify two screens 
in the range, but they do not have to be adjoining). However complicated 
arrangements such as L-shapes may fail. It may also fail if intermediate 
screens are lower resolutions.

The test version is in SVN:  http://csshx.googlecode.com/svn/trunk/csshX

Since I only have one screen, this is very untested. Feedback, and screen-shots 
of screw-ups would be greatly appreciated.

Gavin
```

Original comment by `gavin.brock` on 10 Jun 2010 at 1:20
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Excellent addition, but it doesn't seem to work:

Can't call method "i" without a package or object reference at 
/opt/local/bin/csshX line 590.

```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 1:38

**Comment by GoogleCodeExporter** [comment]
```
Oops - embarrassing. Loose nut on the keyboard.

Fixed in SVN
```

Original comment by `gavin.brock` on 10 Jun 2010 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
Better, but it seems my current setup (MacBook Pro 13" connected to a 22") is 
not being detected correctly:

No such screen [2], screen must be 1 or less at /opt/local/bin/csshX line 590.

The 22" is above the 13", as opposed to them being side by side.
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 1:59

**Comment by GoogleCodeExporter** [comment]
```
Screens are indexed from 0 - try 0-1

If that's the issue, I should document it (or add one!)
```

Original comment by `gavin.brock` on 10 Jun 2010 at 2:04

**Comment by GoogleCodeExporter** [comment]
```
Ok, now it seems the command is right (might pay to change the --help blurb to 
inform that it starts from 0).

Unfortunately it doesn't work for me:
Undefined subroutine &CsshX::Window::max called at /opt/local/bin/csshX line 
607.
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 2:09

**Comment by GoogleCodeExporter** [comment]
```
Ok - screens are now now numbered from 1 (might break some users sorry), but I 
think that is how apple shows them when you identify them in Screen Prefs. I 
documented this.

I fixed the min/max bug (I think).

Thanks for the patience...

```

Original comment by `gavin.brock` on 10 Jun 2010 at 2:52

**Comment by GoogleCodeExporter** [comment]
```
Downloaded the latest version from SVN, but it doesn't seem to work.

If I run with --screen 1-2 I get this:

Argument "1-2" isn't numeric in subtraction (-) at /opt/local/bin/csshX line 
584.
No such screen [2], screen must be 1 or less at /opt/local/bin/csshX line 591.

If I run with --screen 0-1 I get this:

Argument "0-1" isn't numeric in subtraction (-) at /opt/local/bin/csshX line 
584.

PerlObjCBridge: NSException raised while sending objectAtIndex: to NSCFArray 
object
    name:     "NSRangeException"
    reason:   "*** -[NSCFArray objectAtIndex:]: index (-1 (or possibly larger)) beyond bounds (2)"
    userInfo: ""
    location: "/opt/local/bin/csshX line 584"
**** PerlObjCBridge: dying due to NSException

What next?
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 3:06

**Comment by GoogleCodeExporter** [comment]
```
Back from lunch - another SVN update with less stupid bugs....
```

Original comment by `gavin.brock` on 10 Jun 2010 at 3:49

**Comment by GoogleCodeExporter** [comment]
```
Seems to work now. Does a weird thing on the bottom screen with a margin, but 
possibly has to do with sizing the window to make sure it is the same size as 
the ones at the top.
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 3:58

Attachments:
- [Screen shot 2010-06-10 at 3.54.jpg](https://storage.googleapis.com/google-code-attachments/csshx/issue-18/comment-18/Screen shot 2010-06-10 at 3.54.jpg)

**Comment by GoogleCodeExporter** [comment]
```
I made myself unclear in the last sentence. On a second look, I've noticed that 
the bottom windows shifted up in the first (top) screen by the same amount of 
pixels there is on that blank space at the top of the second (bottom) window. 
Probably OS X preventing the window from spanning two screens.
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 4:02

**Comment by GoogleCodeExporter** [comment]
```
Great screenshot - I see what you mean. Does look like the OS, or Terminal.app 
is trying to be clever.

Does the same thing happen if you retile (Ctrl-a r)? How about if you use 
bounds mode (Ctrl-a b) to cover the two windows?
```

Original comment by `gavin.brock` on 10 Jun 2010 at 4:25

**Comment by GoogleCodeExporter** [comment]
```
Bounds mode won't let me resize the window to span more than one screen.

Retiling doesn't change anything either.
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 4:29

**Comment by GoogleCodeExporter** [comment]
```
Ok - I added an extra "setOrigin" call in the grid function. Let's see if that 
helps.

Has csshX ever tiled across the screen borders? If not, I'm going to have to 
find a system to test this on.
```

Original comment by `gavin.brock` on 10 Jun 2010 at 4:32

**Comment by GoogleCodeExporter** [comment]
```
Sorry, no changes.

csshX never tiled across screen borders, so that may be it.
```

Original comment by `rafael.m...@gmail.com` on 10 Jun 2010 at 8:41

**Comment by GoogleCodeExporter** [comment]
```
Just got a Minidisplayport adapter and stole my wife's monitor :-)

Looks like applescript only reliably set the terminal windows location when the 
window is "frontmost".

I rearranged the calls, and now I have windows crossing the monitor border 
(tested on 10.6).

Please let me know if the new SVN version is working for you guys.
```

Original comment by `gavin.brock` on 12 Jun 2010 at 2:23

**Comment by GoogleCodeExporter** [comment]
```
Works for me as well. Great stuff!
```

Original comment by `rafael.m...@gmail.com` on 13 Jun 2010 at 9:01

**Comment by GoogleCodeExporter** [comment]
Original comment by `gavin.brock` on 11 Aug 2010 at 12:34
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Released in 0.73
```

Original comment by `gavin.brock` on 8 Oct 2010 at 1:26
- Changed state: **Done**
